src/lightning/data/streaming/item_loader.py

```diff
@@ -131,8 +135,16 @@ def generate_intervals(self) -> List[Tuple[int, int]]:
         return self._intervals
 
     def load_item_from_chunk(self, index: int, chunk_index: int, chunk_filepath: str, begin: int) -> torch.Tensor:
-        while not os.path.exists(chunk_filepath):
-            sleep(0.0001)
+        if chunk_filepath in self._chunk_filepaths and not os.path.exists(chunk_filepath):
+            del self._chunk_filepaths[chunk_filepath]
+
+        if chunk_filepath not in self._chunk_filepaths:
+            while not os.path.exists(chunk_filepath):
+                sleep(0.001)
+
+            # Wait to avoid any corruption when the file appears
+            sleep(0.001)
```

Comment:
We can probably keep these `sleep`s longer, e.g. `0.01`, as they only occur when `chunk_filepath not in self._chunk_filepaths`

Reply:
done !